### PR TITLE
Return error 400 on invalid new_edits value

### PR DESCRIPTION
**Pull request by eiri**
## Overview

End-point `_bulk_docs` crashes if provided with non-boolean value for `new_edits` parameter.

This patch changes it to return "Bad Request" error instead.

## Testing recommendations

`make javascript suites=bulk_docs`

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

File: test/javascript/tests/bulk_docs.js [hunk 1]
```diff
@@ -110,6 +110,29 @@ couchTests.bulk_docs = function(debug) {
...
+
+  // verify that sending a request with invalid `docs` causes error
+  var req = CouchDB.request("POST", "/" + db_name + "/_bulk_docs", {
+    body: JSON.stringify({"docs": "foo"})
+  });
+
+  T(req.status == 400);
+  result = JSON.parse(req.responseText);
+  T(result.error == "bad_request");
+  T(result.reason == "`docs` parameter must be an array.");
```
Comment (kxepal) [inline test/javascript/tests/bulk_docs.js]: This test looks unrelated to code change. Shouldn't it come with separate commit?

Comment (eiri) [reply test/javascript/tests/bulk_docs.js]: @kxepal I've split the commit in two.

Comment (kxepal) [reply test/javascript/tests/bulk_docs.js]: Thanks!
